test/SlowTests/Issues/RavenDB-16556.cs

```diff
@@ -0,0 +1,275 @@
+
+            var res = session.Advanced
+                .RawQuery<Dto>(query)
+                .WaitForNonStaleResults().ToList();
+            
+            var expected = new List<Dto>{new(){Value = 2, Key = "companies/2-A"}, new(){Value = 1, Key = "companies/1-A"}};
+            
+            Assert.Equal(expected[0].Key, res[0].Key);
+            Assert.Equal(expected[1].Key, res[1].Key);
+            
+            Assert.Equal(expected[0].Value, res[0].Value);
+            Assert.Equal(expected[1].Value, res[1].Value);
+        }
+    }
+
+    [RavenTheory(RavenTestCategory.Querying)]
+    [InlineData(@"from Orders as o
+                                        group by o.CompId
+                                        order by count() as long desc
+                                        load o.EmpId as e
+                                        select 
+                                        {
+                                            Key: key(),
+                                            Value: sum()
+                                        }")]
+    [InlineData(@"declare function 
+                    output(o){
+                        return {
+		                    Key: key(),
+		                    Value: sum()
+                        }
+                    }
+
+                from Orders as o
+                group by o.CompId
+                order by count() as long desc
+                load o.EmpId as e
+                select output(o)")]
+    [InlineData(@"declare function 
+                    output(o){
+                    var empId = key().EmpId;
+                        return {
+		                    Key: empId,
+		                    Value: sum()
+                        }
+                    }
+
+                from Orders as o
+                group by o.CompId
+                order by count() as long desc
+                load o.EmpId as e
+                select output(o)")]
+    public void CheckIfInvalidQueryExceptionIsThrownForIncorrectNumberOfArgsInSum(string query)
+    {
+        using var store = GetDocumentStore();
+
+        using (var session = store.OpenSession())
+        {
+            Employee e1 = new() { Name = "Maciej" };
+            Employee e2 = new() { Name = "Michal" };
+
+            Company c1 = new() { Name = "Company1" };
+            Company c2 = new() { Name = "Company2" };
+
+            session.Store(e1);
+            session.Store(e2);
+
+            session.Store(c1);
+            session.Store(c2);
+
+            Order o1 = new() { EmpId = e1.Id, CompId = c1.Id, Price = 21 };
+            Order o2 = new() { EmpId = e2.Id, CompId = c2.Id, Price = 37 };
+            Order o3 = new() { EmpId = e1.Id, CompId = c2.Id, Price = 44 };
+
+            session.Store(o1);
+            session.Store(o2);
+            session.Store(o3);
+
+            session.SaveChanges();
+            
+            Assert.Throws<Raven.Client.Exceptions.RavenException>(() =>
```

Comment:
Let's assert the exception message as well to ensure that we're getting the error that we expect

Reply:
Done